kernel/kernel-api/src/main/java/io/delta/kernel/internal/ScanImpl.java

```diff
@@ -190,6 +206,56 @@ public void close() throws IOException {
         };
     }
 
+    private Optional<Predicate> getDataSkippingPredicate() {
+        Optional<Predicate> dataPredicate = getDataFilters();
```

Comment:
Something like this would be cleaner

```java
return getDataFilters().map(dataPredicate ->
    DataSkippingUtils.constructDataFilters(dataPredicate, metadata.getSchema())
)
```

Reply:
+1